nuclear/dependencies.py

```diff
@@ -0,0 +1,85 @@
+import os
+import re
+import subprocess
+import sys
+
+whitespace = re.compile("\s+")
+
+
+def parse_requirements_map(user_tools_path):
+    file_path = os.path.abspath(os.path.join(user_tools_path, "requirements.map"))
+    requirements_map = {}
+
+    with open(file_path, "r") as file:
+        for line in file:
+            line = line.strip()
+
+            if len(line) == 0 or line.startswith("#"):
+                continue
+
+            import_name, package_name = whitespace.split(line.strip())
+            requirements_map[import_name] = package_name
+
+    return requirements_map
+
+
+# https://regexr.com/5le57
+dependency_name_regex = re.compile(r"^(.+?)(===|==|~=|!=|<=|>=|>|<)")
```

Comment:
The first group needs to be lazy, otherwise it captures lines with `===` incorrectly:

![image](https://user-images.githubusercontent.com/5924865/114372427-e0106c00-9bc4-11eb-9db1-181aec89e9c7.png)

Reply:
oh shit! nice catch lol. I spent a _good while_ trying to work out if it did anything and I was still wrong ahaha

Reply:
Yes we definitely worked that out before and had no bugs which confused us...